benchmarks/monitoring_log_benchmark.ipynb

```diff
@@ -0,0 +1,227 @@
+    "                    # Get the event name, can be different formats, this should generally get all of them into right format though\n",
+    "                    event_name = ''.join(l[4:]).split(\".\")[-1]\n",
+    "                    # add formatted event [datetime, event_name] to collection\n",
+    "                    events = events +[[dt, event_name]]       \n",
+    "        # Open the monitor log and parse it\n",
+    "        with open(monitor) as f:\n",
+    "            lines = f.readlines()\n",
+    "        \n",
+    "        # Read Caps of Stats we monitor, each one is parsed and converted to float\n",
+    "        cpu_cap = float(re.sub(\"[^$0-9.]\",\"\", [line for line in lines if 'CPU:' in line][0]))\n",
+    "        mem_cap = float(re.sub(\"[^$0-9.]\",\"\", [line for line in lines if 'Total Memory:' in line][0]))\n",
+    "        disk_cap = float(re.sub(\"[^$0-9.]\",\"\", [line for line in lines if 'Total Disk space:' in line][0]))\n",
+    "\n",
+    "        # Collections for plotly trace\n",
+    "        # cpu usages\n",
+    "        cpu = []\n",
+    "        # mem usages\n",
+    "        mem = []\n",
+    "        # disk usages\n",
+    "        disk = []\n",
+    "        # measurement times\n",
+    "        time = []\n",
+    "        # parse through each line and add it to collection\n",
+    "        # EXAMPLE MONITORING LINE GROUP\n",
+    "        \"\"\"\"\n",
+    "        [Tue Jul 17 04:14:59 UTC 2018]\n",
+    "        * CPU usage: 8.1%\n",
+    "        * Memory usage: 6%\n",
+    "        * Disk usage: 57%\n",
+    "        \"\"\"\n",
+    "        for line in lines:\n",
+    "            if 'CPU usage:' in line:\n",
+    "                # if it's a cpu line parse to float and add to cpu collector\n",
+    "                cpu = cpu + [float(re.sub(\"[^$0-9.]\",\"\", line))]\n",
+    "            elif 'Memory usage:' in line:\n",
+    "                # if it's a mem line parse to float and add to mem collector\n",
+    "                mem = mem + [float(re.sub(\"[^$0-9.]\",\"\", line ))]\n",
+    "            elif 'Disk usage:' in line:\n",
+    "                # if it's a disk line parse to float and add to disk collector\n",
+    "                disk = disk + [float(re.sub(\"[^$0-9.]\",\"\", line))]\n",
+    "            elif '[' in line:\n",
+    "                # if it's a time line parse to datetime object and add to time collector\n",
+    "                time = time + [parser.parse(line.replace(\"[\", '').replace(\"]\", '').strip())]\n",
+    "        # create a dataframe to make adding to plotly easier\n",
+    "        df = pd.DataFrame(data={'cores': cpu, 'memory usage (G)': mem, 'disk usage (G)': disk, 'time':time})\n",
+    "        # create the title- eg \"Usage: 64.0 Cores, 236.0 G Memory, 394.0 G Disk Space\" \n",
+    "        title = \"Usage: \" + str(cpu_cap) + \" Cores, \" + str(mem_cap) + \" G Memory, \" + str(disk_cap) + \" G Disk Space\"  \n",
```

Comment:
Here and elsewhere in this file, I recommend using:
* Units in values for memory and disk space, e.g. "236 GB" instead of "236 G".
* Rounded string integers like "64" instead of "64.0".  Floats don't seem sensible for cores.  And even though it might be reasonable for memory or disk space, fractional values are presumably negligible and rounded values would be slightly easier to read.

Reply:
Fixed